consensus/src/metrics_safety_rules.rs

```diff
@@ -35,16 +38,34 @@ impl MetricsSafetyRules {
     pub fn perform_initialize(&mut self) -> Result<(), Error> {
         let consensus_state = self.consensus_state()?;
         let sr_waypoint = consensus_state.waypoint();
-        let proofs = self
-            .storage
-            .retrieve_epoch_change_proof(sr_waypoint.version())
-            .map_err(|e| {
-                Error::InternalError(format!(
-                    "Unable to retrieve Waypoint state from storage, encountered Error:{}",
-                    e
-                ))
-            })?;
-        self.initialize(&proofs)
+        loop {
+            let proofs = self
+                .storage
+                .retrieve_epoch_change_proof(sr_waypoint.version())
+                .map_err(|e| {
+                    Error::InternalError(format!(
+                        "Unable to retrieve Waypoint state from storage, encountered Error:{}",
+                        e
+                    ))
+                })?;
+            let result = self.initialize(&proofs);
```

Comment:
nit: you could probably just match this directly, e.g.,:

```
match self.initialize(&proofs) {
    Err(Error::Waypoint...) => {...}
    result => return result,
}
```

Would that work?

Reply:
Updated the code to match directly and it works!